content/api/cypress-api/config.md

```diff
@@ -156,8 +156,25 @@ file, using test configuration will only change configuration values during the
 suite or test where they are set. The values will then reset to the previous
 default values after the suite or test is complete.
 
-See the full guide on
-[test configuration](/api/cypress-api/config#Test-Configuration).
+See the full guide on [test-time configuration](/api/cypress-api/config#Test-time-overrides).
+
+## `Cypress.config()` executes Synchronously
+
+It's important to note that `Cypress.config()` executes synchronously and will not wait for the Cypress commands above it to execute.
+If you need to update your configuration mid-test, be sure to either chain the command the
+[asynchronously Cypress command](/guides/core-concepts/introduction-to-cypress#Commands-Are-Asynchronous) before it or wrap execute it with
+the `cy.then` Cypress command to correctly queue.
```

Comment:
We don't recommend chaining `.then` directly off of `cy`. If we decide to enforce chaining in a later version this would cause a lot of issues for people we recommended it to.

```suggestion
If you need to update your configuration mid-test, be sure to either chain the command the
[asynchronously using a Cypress command](/guides/core-concepts/introduction-to-cypress#Commands-Are-Asynchronous) before it.
```